### PR TITLE
feature(form): use matInput for color picker

### PR DESCRIPTION
**Pull request by nappynapster**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser-Support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public tooltipText: string;
+
+    /** Stores the callback function that will be called on blur. */
+    public _onTouchedCallback: () => void = noop;
+    /** Stores the callback function that will be called when the control's value changes in the UI. */
+    public _onChangeCallback: (_: any) => void = noop;
+
+    /** Registers a callback function that is called when the control's value changes in the UI. */
+    public registerOnChange(fn: (_: any) => void): void {
+        // this._onChangeCallback = fn;
```
Comment (Neill83) [inline src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts:41]:
```suggestion
        this._onChangeCallback = fn;
```

File: src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /** Registers a callback function that is called by the forms API on initialization to update the form model on blur. */
+    public registerOnTouched(fn: () => void): void {
+        // this._onTouchedCallback = fn;
```
Comment (Neill83) [inline src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts:46]:
```suggestion
        this._onTouchedCallback = fn;
```

**Review by Neill83 — CHANGES_REQUESTED** [review]

File: src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts (hunk 1 above)
Comment (nappynapster) [reply src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts:41]: habe die sachen bewusst auskommentiert, da hier die component keine Funktionalität beinhalten soll, nur die tests an sich

File: src/lib/components/forms/form/form-components/color-picker/color-picker.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    it('should call registered change callback whenever the value of the input is changed by the user', async () => {
+        const onChangeCallback: jasmine.Spy = jasmine.createSpy('onChange');
+        component.registerOnChange(onChangeCallback);
+
+        await input.setValue(testColor);
+
+        expect(onChangeCallback).toHaveBeenCalledWith(testColor);
+    });
+
+    it('should call registered onTouchedCallback when select blur has been called', async () => {
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/color-picker/color-picker.component.spec.ts:107]:
```suggestion
    it('should call registered onTouchedCallback when input was blurred', async () => {
```

**Review by pweyrich — APPROVED** [review]
Sonst Tip top 👍

**Review by niwolf — APPROVED** [review]
👍

**Review by niwolf — DISMISSED** [review]

File: src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    /** @description The internal data model */
+    public value: string;
+
+    /** hex color pattern to validate color strings */
+    public _pattern: string = TerraRegex.COLOR_HEX;
+
+    /** Stores the callback function that will be called on blur. */
+    public _onTouchedCallback: () => void = noop;
+    /** Stores the callback function that will be called when the control's value changes in the UI. */
+    public _onChangeCallback: (_: any) => void = noop;
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/color-picker/color-picker.component.ts:49]:
```suggestion
    public _onChangeCallback: (_: string) => void = noop;
```

**Review by pweyrich — APPROVED** [review]

**Review by pweyrich — APPROVED** [review]

**Pull request merged**